store.go

```diff
@@ -1444,7 +1445,13 @@ func (s *store) CreateContainer(id string, names []string, image, layer, metadat
 	}
 	container, err := rcstore.Create(id, names, imageID, layer, metadata, options)
 	if err != nil || container == nil {
-		rlstore.Delete(layer)
+		if err2 := rlstore.Delete(layer); err2 != nil {
+			if err == nil {
+				err = fmt.Errorf("deleting layer %#v: %w", layer, err2)
```

Comment:
In past changing error message or introducing new error have regressed `podman` or `buildah` tests. Should we open a dummy PR in podman/buildah to verify these ?

Reply:
I wouldn’t expect Podman tests to trigger _these_ error recovery paths, but, sure, it’s easy enough to create the PRs, and this set of changes is tedious to review, so the extra confidence is definitely worth it.

https://github.com/containers/buildah/pull/4311
https://github.com/containers/podman/pull/16024

cover both this PR and #1364 .

Reply:
Both succeeded.